server.go

```diff
@@ -1372,6 +1375,11 @@ func (s *Server) Shutdown() {
 	// TODO(aditya) shut down workers and socket readers
 	s.logger.Info("Shutting down server gracefully")
 	close(s.shutdown)
+	if s.FlushOnShutdown {
+		ctx, cancel := context.WithDeadline(context.Background(), time.Now().Add(s.Interval))
```

Comment:
is interval required? what happens if flush is set to true, but interval isn't set? could that potentially lead to not allowing enough time for flushing?

Reply:
It looks like veneur has a default interval setting of 10s. My concern with not setting some sort of timeout is that the sidecar might get stuck in some sort of shutdown flow. In the case of flushing taking too long, this timeout also applies to the normal flush to sinks, so if flushing takes longer than this, you're already dropping metrics.